### ISSUE TITLE
Creating Falkor training dataset

### ISSUE THREAD

**Issue by wkumler** [wkumler/MS_metrics#2]
After chatting with Bryna yesterday about my difficulties using the Muller dataset, we've decided to train up my own using my own data. I'm planning to use the Falkor dataset (41 files: 1 blank, 24 samples, 6 pooled, and 10 standards).

I'm optimistic about this method because 1) no real feature-based training data sets exist that I've been able to find and 2) feature-based seems to be much more accurate for what I do when I'm trying to decide whether something is a peak or not and 3) feature-based is much less-well reported on. Also, it would be great to be able to say something like "don't worry so much about your peakpicking parameters and focus instead on design-of-experiments decisions as to whether or not the peak is real"

**Comment by wkumler** [comment]
Some questions I've got floating around as I try to do this:

1. How do I make sure retention time correction occurs?
2. Do we want to sample peaks randomly or comprehensively? 
3. Do we want to intentionally include replacements so we can test within-person accuracy?
4. Which files to include? 
    - I'm inclined to include all of them because they all have different bits of information in them
5. Do we do any quality control ahead of time? 
    - The current `xdata_filled` object has my custom quality metric included but that's not the eventual goal - we want to directly export xcms output into this step and filter that way. How far up the peakpicking quality chain do we go?
    - The less QC we do during the peakpicking step the slower everything will be
6. How do we decide retention time bounds? Min/max of every peak within the feature?
7. Do we plot zeroes?
8. How to pick up where we left off before?
9. Memory problems
    - How do we load only a subset of the chromatograms at one time?

**Comment by wkumler** [comment]
Bryna and I also discussed today some unsupervised clustering methods with my vision being to group features in DoE space and then circle around those that have all the similar features. I.e., calculate difference between blank and samples and select all the clear second peak where samps>>>blank.

**Comment by wkumler** [comment]
How large should the training dataset be? With peakpicking parameters set below we get 350 features, the vast majority of which are trash:

```
cwp <- CentWaveParam(
  ppm = 5, 
  peakwidth = c(20, 80), 
  prefilter = c(5, 1e7), 
  snthresh = 0, 
  verboseColumns = TRUE, 
  extendLengthMSW = TRUE, 
  integrate = 2
)
obp <- ObiwarpParam(
  binSize = 0.1, 
  centerSample = round(nrow(file_data)/2), 
  response = 1, 
  distFun = "cor_opt"
)
pdp <- PeakDensityParam(
  sampleGroups = file_data$colid, 
  bw = 12, 
  minFraction = 0.1, 
  binSize = 0.001, 
  minSamples = 2
)
fpp <- FillChromPeaksParam(ppm = 2.5)
```

**Comment by wkumler** [comment]
After spending a few weeks exploring a training set of 350 features, we've realized that I might just not have the explanatory power we're interested in because we only have 26(!!) good features. This is slightly worse than the 255,000 dataset which had about one good peak for every 3 bad ones.

**Comment by wkumler** [comment]
When dropping the prefilter thresholds to 

```
cwp <- CentWaveParam(
  ppm = 5, 
  peakwidth = c(20, 80), 
  prefilter = c(3, 1e6), 
  snthresh = 0, 
  verboseColumns = TRUE, 
  extendLengthMSW = TRUE, 
  integrate = 2
)
```

we get 2,040 features instead. Time to categorize those too!

**Comment by wkumler** [comment]
Main question: do we plot zeroes? The peaks often look wildly worse when they're included and it's more like a peak made up of spikes instead of an actual peak. When plotting without zeroes it's more like a silhouette of a peak and looks much better.

This peak looks good (and probably is good):
![image](https://user-images.githubusercontent.com/43116923/215892261-76da1a57-1de8-4a7c-92b5-7203b5938125.png)

This peak looks bad (and probably is good):
![image](https://user-images.githubusercontent.com/43116923/215892341-5e672c90-a32b-48e1-9b40-7d82d09a188b.png)

despite being the SAME EXACT PEAK. Maybe HILIC data just has too many missed scans to be useful? Or my m/z window is too small? Quite possibly the second one tbh even though these are the theoretical boundaries on the peak that xcms detects, not any particular ppm window.

**Comment by wkumler** [comment]
Update: I was getting so many missed scans because of how I was defining the bounding boxes. I had a bug in the code where I used the max of the LOWER bounds as the overall upper bound instead of the max of the UPPER bounds. See [179e76f](https://github.com/wkumler/MS_metrics/commit/179e76f422a4306270940076af33d6639b81432f) and [ffd7f00](https://github.com/wkumler/MS_metrics/commit/ffd7f002969ef69b1884737705323f1ff4253b76). With this, I think the training set looks pretty good so I'm closing this issue.
